server/service/labels.go

```diff
@@ -186,37 +185,20 @@ func (svc *Service) ModifyLabel(ctx context.Context, id uint, payload fleet.Modi
 		return nil, nil, fleet.NewInvalidArgumentError("hosts", "cannot provide a list of hosts for a dynamic label")
 	}
 
-	// if membership type is manual and the Hosts membership is provided, must
-	// use ApplyLabelSpecs (as SaveLabel does not update label memberships),
-	// otherwise SaveLabel works for dynamic membership. Must resolve the host
-	// identifiers to hostname so that ApplySpecs can be used (it expects only
-	// hostnames).
-	if label.LabelMembershipType == fleet.LabelMembershipTypeManual && payload.Hosts != nil {
-		spec := fleet.LabelSpec{
-			Name:                originalLabelName,
-			Description:         label.Description,
-			Query:               label.Query,
-			Platform:            label.Platform,
-			LabelType:           label.LabelType,
-			LabelMembershipType: label.LabelMembershipType,
-		}
-		hostnames, err := svc.ds.HostnamesByIdentifiers(ctx, payload.Hosts)
+	// use SaveLabel to update label info, and UpdateLabelMembershipByHostIDs to update membership. Approach using label
+	// names and ApplyLabelSpecs doesn't work for multiple hosts with the same name.
+
+	if payload.Hosts != nil {
+		// get host ids for valid hosts. since this endpoint will contain hosts identified by serial
+		// number, there should be no duplicates
+
+		hostIds, err := svc.ds.HostIDsByIdentifier(ctx, filter, payload.Hosts)
```

Comment:
The only caveat with this is that it searches fewer fields.  `HostnamesByIdentifiers` matches against:
* hostname
* osquery_host_id
* node_key
* uuid
* hardware_serial

while `HostIDsByIdentifier` only matches against:
* hostname
* uuid
* hardware_serial

This is probably just due to different use cases at the time, but we'll want the new API to support the same as the old so we should update `HostnamesByIdentifiers` to match, or add options to the method indicating what to match against.  I can handle it in my follow-on.

Reply:
Sounds good